yjit/src/codegen.rs

```diff
@@ -7774,19 +7774,19 @@ fn gen_getblockparamproxy(
         return Some(EndBlock);
     }
 
-    // A mirror of the interpreter code. Checking for the case
-    // where it's pushing rb_block_param_proxy.
-
     // EP level
     let level = jit.get_arg(1).as_u32();
 
     // Peek at the block handler so we can check whether it's nil
     let comptime_handler = jit.peek_at_block_handler(level);
 
-    // When a block handler is present, it should always be a GC-guarded
-    // pointer (VM_BH_ISEQ_BLOCK_P)
-    if comptime_handler.as_u64() != 0 && comptime_handler.as_u64() & 0x3 != 0x1 {
-        incr_counter!(gbpp_not_gc_guarded);
```

Comment:
Because this was incrementing the counter once at compile time, it wasn't counting when the generated code was exiting through this path. That was the cause for the [breakdown counters not summing to the total](https://github.com/Shopify/ruby/issues/524#issuecomment-1652100626).